File: common/src/main/java/generations/gg/generations/core/generationscore/world/item/legends/ElementalPostBattleUpdateItemImpl.java
package generations.gg.generations.core.generationscore.world.item.legends;

import com.cobblemon.mod.common.api.types.ElementalType;
import com.cobblemon.mod.common.battles.actor.PlayerBattleActor;
import net.minecraft.world.item.ItemStack;

import java.util.List;

public class ElementalPostBattleUpdateItemImpl extends ElementalPostBattleUpdateItem {
    private final List<ElementalType> types;

    public ElementalPostBattleUpdateItemImpl(Properties properties, ElementalType... types) {
        super(properties);
        this.types = List.of(types);
    }

    @Override
    public boolean checkType(PlayerBattleActor actor, ItemStack stack, ElementalType type) {
        return types.contains(type);
    }
}

File: src/common/main/kotlin/generations/gg/generations/core/generationscore/common/world/item/legends/ElementalPostBattleUpdateItemImpl.kt
package generations.gg.generations.core.generationscore.common.world.item.legends

import com.cobblemon.mod.common.api.types.ElementalType
import com.cobblemon.mod.common.battles.actor.PlayerBattleActor
import net.minecraft.world.item.ItemStack

open class ElementalPostBattleUpdateItemImpl(properties: Properties?, private val types: List<ElementalType>) :
    ElementalPostBattleUpdateItem(properties) {
    constructor(properties: Properties?, type: ElementalType) : this(properties, java.util.List.of<ElementalType>(type))

    override fun checkType(actor: PlayerBattleActor, stack: ItemStack, type: ElementalType): Boolean {
        return types.contains(type)
    }
}
